File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, m: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        k=0
        for i in range(len(m)):
            for j in range(k,len(m)):
                m[i][j],m[j][i]=m[j][i],m[i][j]
            k+=1
        for i in range(len(m)):
            m[i].reverse()
        

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& m) 
    {
        int i,j;
        int x=0;
        int k=0;
        for(i=0;i<m.size();i++)
        {
            for(j=0+k;j<m[i].size();j++)
            {
                swap(m[i][j],m[j][i]);
            }
            k+=1;
        }
        for(i=0;i<m.size();i++)
        {
            reverse(m[i].begin(),m[i].end());
        }
    }
};